overlord/devicestate/devicemgr.go

```diff
@@ -1369,6 +1370,18 @@ func (m *DeviceManager) hasFDESetupHook() (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("cannot get kernel info: %v", err)
 	}
+	return hasFDESetupHookInKernel(kernelInfo), nil
+}
+
+func hasFDESetupHookInKernel(kernelInfo *snap.Info) bool {
 	_, ok := kernelInfo.Hooks["fde-setup"]
-	return ok, nil
+	return ok
+}
+
+func checkFDEFeatures(st *state.State, kernelInfo *snap.Info) error {
+	// TODO: run the fde-setup hook with "op":"features".  If the
+	//       hooks returns {"features":[]} we consider the
+	//       hardware supported. If the hook errors or if it
+	//       returns {"error":"hardware unsupported"} we don't.
```

Comment:
might idea was for that error to be machine readable with a hyphen: hardware-unsupported